### PR TITLE
bumped go to 1.21 and created helm chart

### LINKED ISSUE #19: bump dependencies and go to 1.21

### PR DESCRIPTION
**Pull request by mzeevi**
Fixes #19, #20, #21. With this PR, the go version has been upgraded. In addition, some linting issues have been fixed and a helm chart has been created using helmify

### CODE REVIEW

**Review by liadlauber — APPROVED** [review]
/lgtm

File: .github/workflows/pre-merge.yaml [hunk 1]
```diff
@@ -22,6 +21,11 @@ jobs:
         uses: actions/setup-go@v4
         with:
           go-version-file: go.mod
+      - name: Run golangci-lint
```
Comment (omer5yerushalmi) [inline .github/workflows/pre-merge.yaml:24]: Need to add .golangci.yml as a configuration file for the CI

**Review by omer5yerushalmi — APPROVED** [review]

**Review by omer5yerushalmi — APPROVED** [review]

**Review by omer5yerushalmi — APPROVED** [review]

**Pull request merged**